### ISSUE TITLE
bugreport

### ISSUE THREAD

**Issue by aiman-waheed** [aiman-waheed/liabraray-management-systm-#1]
# diabetes_ml.py

import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

data = pd.read_csv("diabetes.csv")

# Optional: first 5 rows dekhne ke liye
print("First 5 rows:")
print(data.head())

# 2) Features (X) aur label (y) separate karo
X = data.drop("Outcome", axis=1)   # saare columns except Outcome
y = data["Outcome"]                # target column

# 3) Train-test split (80% train, 20% test)
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

# 4) Feature scaling (important for ML algorithms)
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# 5) Model select karo (Logistic Regression – simple & common)
model = LogisticRegression(max_iter=1000)

# 6) Model train karo
model.fit(X_train_scaled, y_train)

# 7) Test data pe predict karo
y_pred = model.predict(X_test_scaled)

# 8) Accuracy check karo
acc = accuracy_score(y_test, y_pred)
print(f"\nAccuracy on test set: {acc * 100:.2f}%")

# 9) Confusion matrix & detailed report
print("\nConfusion Matrix:")
print(confusion_matrix(y_test, y_pred))

print("\nClassification Report:")
print(classification_report(y_test, y_pred))

# 10) Example: new patient ki prediction
# Order of features:
# [Pregnancies, Glucose, BloodPressure, SkinThickness, Insulin, BMI,
#  DiabetesPedigreeFunction, Age]

new_patient = [[2, 120, 70, 25, 80, 28.5, 0.35, 30]]

new_patient_scaled = scaler.transform(new_patient)
prediction = model.predict(new_patient_scaled)[0]
prob = model.predict_proba(new_patient_scaled)[0][1]

print("\nNew patient prediction:")
print("Has diabetes" if prediction == 1 else "No diabetes")
print(f"Probability of diabetes: {prob * 100:.2f}%")
